src/merkle_set.rs

```diff
@@ -114,6 +116,22 @@ fn radix_sort(range: &mut [[u8; 32]], depth: u8) -> ([u8; 32], NodeType) {
                 (child_hash, child_type)
             }
         }
+    } else if depth == 255 {
+        // this is an edge case where we make it all the way down to the
+        // bottom of the tree, and split the last pair. This has the same
+        // effect as the else-block, but since we use u8 for depth, it would
+        // overflow
```

Comment:
```suggestion
        // overflow. Thus, this case must be tested before the code in the below else-block.
```

This is a great comment. It is implied, but it might be nice to explicitly state that this block needs to come before the `else` case

Reply:
I'm not quite sure I follow your addition. The edge case is run instead of the else-block, the order doesn't really matter. you could put the else block first and have the condition `depth < 255`. But perhaps I'm misunderstanding

Reply:
No, you are correct; what I meant is equivalent to putting `debug_assert!(depth < 255);` in the else block.

Totally not needed.